OsmAnd-java/src/main/java/net/osmand/GPXUtilities.java

```diff
@@ -848,7 +848,8 @@ public GPXTrackAnalysis prepareInformation(long filestamp, SplitSegment... split
 						totalDistance += calculations[0];
 						segmentDistance += calculations[0];
 						point.distance = segmentDistance;
-						timeDiff = (int)((point.time - prev.time) / 1000);
+						long timeDiffMillis = Math.abs(point.time - prev.time);
```

Comment:
Probably here we shoud use Math.max(0, point.time - prev.time)

Otherwise we could get ridiculous numbers... 
Or probably even better to have prevTime variable equals prevTime = Math.max(point.time, prevTime);